src/agent/debuglet.ts

```diff
@@ -322,6 +349,32 @@ export class Debuglet extends EventEmitter {
   }
 
   /**
+   * isReady returns a promise that only resolved if the last breakpoint update
+   * happend within a duration (PROMISE_RESOLVE_CUT_OFF_IN_MILLISECONDS). This
+   * feature is mainly used in Google Cloud Function (GCF), as it is a
+   * serverless environment and we wanted to make sure debug agent always
+   * captures the snapshots.
+   */
+  async isReady(): Promise<void> {
+    if (Date.now() < this.breakpointFetchedTimestamp +
+            PROMISE_RESOLVE_CUT_OFF_IN_MILLISECONDS) {
+      return Promise.resolve();
+    } else {
+      if (this.breakpointFetched) {
+        return this.breakpointFetched;
+      }
+      this.debuggeeRegistered.then(() => {
+        this.breakpointFetched = new Promise<void>((resolve) => {
+          this.breakpointFetchedResolve = resolve;
+        });
+        this.scheduleBreakpointFetch_(
+            0 /*immediately*/, true /*only fetch once*/);
+        return this.breakpointFetched;
+      });
```

Comment:
It looks like at this point the function doesn't return anything, which is automatically converted to a `Promise<void>` since the function is marked `async`.  However, the returned `Promise<void>` is not `this.breakpointFetched` and so the code won't work as expected.

Reply:
Checked back on this. I think separating listBreakpoint out would be tricky because it also calls scheduleBreakpointFetch for regular scheduling. Instead, I have made change in isReady to make sure the correct promise (breakpointFetched) is returned.